cedar-policy-core/src/est.rs

```diff
@@ -379,7 +379,7 @@ impl From<ast::Template> for Policy {
 
 impl<T: Clone> From<ast::Expr<T>> for Clause {
     fn from(expr: ast::Expr<T>) -> Clause {
-        Clause::When(expr.into())
+        Clause::When(ast::Expr::<T>::into_expr::<T, Builder>(expr))
     }
```

Comment:
I think we should be able to make this `expr.into_expr()` if we remove the `V` parameter from `into_expr()`?